src/categorical_algebra/CSets.jl

```diff
@@ -1083,6 +1106,22 @@ function unpack_components(αs::AbstractVector{<:StructACSetTransformation{S}};
   NamedTuple(c => map(α -> α[c], αs) for c in names)
 end
 
+"""
+Every (non-empty) diagram has a representative object we can use to get  
+dynamic type information from
+"""
+diagram_ob(m::Multispan{DynamicACSet}) = apex(m)
```

Comment:
Since it doesn't matter which object this is (as long as the user hasn't mixed up their acsets!), I think you just define this as `first(cocone_objects(d))` for all colimits, which would fail only for empty diagrams.